### PR TITLE
Make job_runner calculate the Status, pass the status object to the reporter interface

### PR DESCRIPTION
**Pull request by marcoguerri**
Still working in progress, tests are broken, but gives an idea of how the status structures should be used in the reporter interface.

### CODE REVIEW

File: pkg/runner/job_status.go [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+func (jr *JobRunner) BuildRunStatuses(currentJob *job.Job) ([]job.RunStatus, error) {
+
+	runStatuses := make([]job.RunStatus, 0, currentJob.Runs)
+	for runID := uint(0); runID < currentJob.Runs; runID++ {
+		runCoordinates := RunCoordinates{JobID: currentJob.ID, RunID: types.RunID(runID)}
+		runStatus, err := jr.BuildRunStatus(runCoordinates, currentJob)
+		if err != nil {
+			return nil, fmt.Errorf("could not rebuild run status for run %d: %v", runID, err)
+		}
+		runStatuses[runID] = *runStatus
```
Comment (insomniacslk) [inline pkg/runner/job_status.go]: this seems to fail for me on a 3-runs job:
```
panic: runtime error: index out of range [0] with length 0
```

Just change line 184:
```
runStatuses := make([]job.RunStatus, 0, currentJob.Runs)
```
to
```
runStatuses := make([]job.RunStatus, currentJob.Runs)
```
so it creates a slice with `Runs` elements

Comment (marcoguerri) [reply pkg/runner/job_status.go]: Indeed, thanks, I barely tested this last night. I think I'll follow the same approach as the other control paths and `append` instead. Performance-wise should be the same?

**Comment by tleunen** [comment]
# [Codecov](https://codecov.io/gh/facebookincubator/contest/pull/55?src=pr&el=h1) Report
> Merging [#55](https://codecov.io/gh/facebookincubator/contest/pull/55?src=pr&el=desc) into [master](https://codecov.io/gh/facebookincubator/contest/commit/bc32969f2bca3d80f1d09fa37deed42fe0b0a634&el=desc) will **decrease** coverage by `31.47%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/facebookincubator/contest/pull/55?src=pr&el=tree)

```diff
@@             Coverage Diff             @@
##           master      #55       +/-   ##
===========================================
- Coverage   61.14%   29.66%   -31.48%     
===========================================
  Files          42        9       -33     
  Lines        2473      445     -2028     
===========================================
- Hits         1512      132     -1380     
+ Misses        748      302      -446     
+ Partials      213       11      -202     
```

| Flag | Coverage Δ | |
|---|---|---|
| #integration | `?` | |
| #unittests | `29.66% <0.00%> (+3.72%)` | :arrow_up: |

| [Impacted Files](https://codecov.io/gh/facebookincubator/contest/pull/55?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [pkg/pluginregistry/pluginregistry.go](https://codecov.io/gh/facebookincubator/contest/pull/55/diff?src=pr&el=tree#diff-cGtnL3BsdWdpbnJlZ2lzdHJ5L3BsdWdpbnJlZ2lzdHJ5Lmdv) | `18.42% <0.00%> (-65.79%)` | :arrow_down: |
| [pkg/pluginregistry/bundles.go](https://codecov.io/gh/facebookincubator/contest/pull/55/diff?src=pr&el=tree#diff-cGtnL3BsdWdpbnJlZ2lzdHJ5L2J1bmRsZXMuZ28=) | `0.00% <0.00%> (-53.97%)` | :arrow_down: |
| [pkg/lib/comparison/comparison.go](https://codecov.io/gh/facebookincubator/contest/pull/55/diff?src=pr&el=tree#diff-cGtnL2xpYi9jb21wYXJpc29uL2NvbXBhcmlzb24uZ28=) | `36.00% <0.00%> (-29.34%)` | :arrow_down: |
| [pkg/test/parameter.go](https://codecov.io/gh/facebookincubator/contest/pull/55/diff?src=pr&el=tree#diff-cGtnL3Rlc3QvcGFyYW1ldGVyLmdv) | `50.00% <0.00%> (-21.43%)` | :arrow_down: |
| [pkg/test/step.go](https://codecov.io/gh/facebookincubator/contest/pull/55/diff?src=pr&el=tree#diff-cGtnL3Rlc3Qvc3RlcC5nbw==) | `0.00% <0.00%> (-20.00%)` | :arrow_down: |
| [plugins/targetlocker/inmemory/inmemory.go](https://codecov.io/gh/facebookincubator/contest/pull/55/diff?src=pr&el=tree#diff-cGx1Z2lucy90YXJnZXRsb2NrZXIvaW5tZW1vcnkvaW5tZW1vcnkuZ28=) | `47.95% <0.00%> (-18.37%)` | :arrow_down: |
| [pkg/test/result.go](https://codecov.io/gh/facebookincubator/contest/pull/55/diff?src=pr&el=tree#diff-cGtnL3Rlc3QvcmVzdWx0Lmdv) | `0.00% <0.00%> (-58.07%)` | :arrow_down: |
| [pkg/storage/jobs.go](https://codecov.io/gh/facebookincubator/contest/pull/55/diff?src=pr&el=tree#diff-cGtnL3N0b3JhZ2Uvam9icy5nbw==) | | |
| [pkg/storage/events.go](https://codecov.io/gh/facebookincubator/contest/pull/55/diff?src=pr&el=tree#diff-cGtnL3N0b3JhZ2UvZXZlbnRzLmdv) | | |
| [plugins/storage/memory/memory.go](https://codecov.io/gh/facebookincubator/contest/pull/55/diff?src=pr&el=tree#diff-cGx1Z2lucy9zdG9yYWdlL21lbW9yeS9tZW1vcnkuZ28=) | | |
| ... and [23 more](https://codecov.io/gh/facebookincubator/contest/pull/55/diff?src=pr&el=tree-more) | |

------

[Continue to review full report at Codecov](https://codecov.io/gh/facebookincubator/contest/pull/55?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/facebookincubator/contest/pull/55?src=pr&el=footer). Last update [bc32969...3df6fe8](https://codecov.io/gh/facebookincubator/contest/pull/55?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

File: pkg/runner/job_status.go (hunk 1 above)
Comment (marcoguerri) [reply pkg/runner/job_status.go]: Fixed.

**Comment by tleunen** [comment]
# [Codecov](https://codecov.io/gh/facebookincubator/contest/pull/55?src=pr&el=h1) Report
> Merging [#55](https://codecov.io/gh/facebookincubator/contest/pull/55?src=pr&el=desc) into [master](https://codecov.io/gh/facebookincubator/contest/commit/bc32969f2bca3d80f1d09fa37deed42fe0b0a634&el=desc) will **increase** coverage by `2.52%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/facebookincubator/contest/pull/55?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master      #55      +/-   ##
==========================================
+ Coverage   61.14%   63.66%   +2.52%     
==========================================
  Files          42       42              
  Lines        2473     2518      +45     
==========================================
+ Hits         1512     1603      +91     
+ Misses        748      695      -53     
- Partials      213      220       +7     
```

| Flag | Coverage Δ | |
|---|---|---|
| #integration | `62.10% <66.99%> (+2.56%)` | :arrow_up: |
| #unittests | `29.66% <0.00%> (+3.72%)` | :arrow_up: |

| [Impacted Files](https://codecov.io/gh/facebookincubator/contest/pull/55?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [pkg/test/result.go](https://codecov.io/gh/facebookincubator/contest/pull/55/diff?src=pr&el=tree#diff-cGtnL3Rlc3QvcmVzdWx0Lmdv) | `0.00% <0.00%> (-58.07%)` | :arrow_down: |
| [pkg/runner/job\_status.go](https://codecov.io/gh/facebookincubator/contest/pull/55/diff?src=pr&el=tree#diff-cGtnL3J1bm5lci9qb2Jfc3RhdHVzLmdv) | `65.81% <0.00%> (ø)` | |
| [plugins/storage/memory/memory.go](https://codecov.io/gh/facebookincubator/contest/pull/55/diff?src=pr&el=tree#diff-cGx1Z2lucy9zdG9yYWdlL21lbW9yeS9tZW1vcnkuZ28=) | `92.72% <0.00%> (+1.81%)` | :arrow_up: |
| [plugins/storage/rdbms/events.go](https://codecov.io/gh/facebookincubator/contest/pull/55/diff?src=pr&el=tree#diff-cGx1Z2lucy9zdG9yYWdlL3JkYm1zL2V2ZW50cy5nbw==) | `76.61% <0.00%> (+2.41%)` | :arrow_up: |
| [pkg/event/testevent/test.go](https://codecov.io/gh/facebookincubator/contest/pull/55/diff?src=pr&el=tree#diff-cGtnL2V2ZW50L3Rlc3RldmVudC90ZXN0Lmdv) | `88.88% <0.00%> (+11.11%)` | :arrow_up: |
| [pkg/storage/events.go](https://codecov.io/gh/facebookincubator/contest/pull/55/diff?src=pr&el=tree#diff-cGtnL3N0b3JhZ2UvZXZlbnRzLmdv) | `77.14% <0.00%> (+20.00%)` | :arrow_up: |

------

[Continue to review full report at Codecov](https://codecov.io/gh/facebookincubator/contest/pull/55?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/facebookincubator/contest/pull/55?src=pr&el=footer). Last update [bc32969...42ec8ad](https://codecov.io/gh/facebookincubator/contest/pull/55?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Pull request merged**